### DISCUSSION TITLE
RaspberryMatic 3.73.9.20240130

### DISCUSSION THREAD

**Discussion by jens-maus** [OpenCCU/OpenCCU#2633] [Announcements]
This is release 3.73.9.20240130 of RaspberryMatic which is a **major** release with the following bugfixes and feature changes:

[![Downloads](https://img.shields.io/github/downloads/jens-maus/RaspberryMatic/3.73.9.20240130/total.svg?style=flat-square)](https://github.com/jens-maus/RaspberryMatic/releases/3.73.9.20240130) ![License](https://img.shields.io/github/license/jens-maus/RaspberryMatic.svg?style=flat-square) [![Donate](https://img.shields.io/badge/donate-PayPal-green.svg?style=flat-square)](https://www.paypal.com/cgi-bin/webscr?cmd=_s-xclick&hosted_button_id=RAQSDY9YNZVCL) [![GitHub sponsors](https://img.shields.io/static/v1?label=Sponsor&message=%E2%9D%A4&logo=GitHub&link=https://github.com/sponsors/jens-maus)](https://github.com/sponsors/jens-maus)

<sub>[Hier klicken für deutschsprachige 🇩🇪🇦🇹🇨🇭 Version des ChangeLogs/Diskussionsbeitrag](https://homematic-forum.de/forum/viewtopic.php?f=65&t=81602)</sub>

## 🚧 Changes:
<sub>For all changes, see the [full commit log](https://github.com/jens-maus/RaspberryMatic/compare/3.73.9.20231130...3.73.9.20240130).</sub>

#### CCU/HomeMatic service changes:
- update [OCCU](https://github.com/eq-3/occu) version to [3.73.9-2](https://github.com/jens-maus/occu/tree/b_3_73) which comes with the following changes:
   - update `ReGaHss` logic engine to version `R1.00.0388.0236 (Jan 16 2024)`:
      - fix problem in `StrValueByIndex()` where upon specifying an index < 0 the function returned different results. Now an empty string is returned and a warning will be output in the log accordingly (#2597). 
      - fix `ChannelType()` to allow `ictXXX` values to be passed as well as plain integer numbers and to correctly return true/false (#2605).
      - fix segfault which occurred when `Name()` was called on `OT_DEVICES` objects for renaming this enum-based object (#2601).
      - fix floating point exception issues with `Random()` function which occurred when min > max in the `Random()` call and max is one smaller than min. In addition, `Random()` was returning strange results in case the min value was larger than max (#2598).
- add full fledged hardware support for the new RaspberryPi5 for which a dedicated image is now distributed alongside the other RaspberryPi based platforms (#2506).
- add new `generic-aarch64` platform and install images which can be primarily used for running RaspberryMatic as a virtual appliance under an ARM-based virtualization platform like [Proxmox VE for ARM](https://github.com/jiangcuo/Proxmox-Port). That means, a RaspberryMatic based virtual appliance can now be installed, e.g. in a Proxmox VE environment which is running on ARM-based systems like a RaspberryPi4, RaspberryPi5 and similar powerful ARM-based hardware.  To ease installation of such ARM-based virtualization environments, the `install-proxmox.sh` helper script was updated as well as a dedicated [Proxmox VE installation documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE#proxmox-ve-arm-grundinstallation) added (#2627).

#### WebUI changes:
- add new `0183-WebUI-ImprovedFirmwareupdateDialog` WebUI patch to display a slightly improved device firmware update dialog with a hided serial number display (#2532, @MichaelN0815, @Baxxy13).
- integrate fix in `0137-WebUI-Add-NoInterfaceWarning` WebUI patch in case no interface is present but a device is about to enter the firmware check mechanism.
- add new `0184-WebUI-Fix-EM8LockedChannelBehaviour` WebUI patch fixing locked channel behaviour for `HM-MOD-EM8` (#2599, @Baxxy13, @jp112sdl).
- rework `0130-WebUI-Fix-ChannelSysVarPrefix` WebUI patch to show 'optical signal' prefixes also for HmIP-ASIR (#2589, @Baxxy13).
- update CodeMirror JS Editor framework to latest 5.65.16.
- remove port 8443 use for https device check urls to minimize local issues with blocked non-standard ports and similar.

#### Operating system changes:
- fix error output in `S02InitRTC` in case of HA addon and RPI-RF-MOD GPIO use.
- add empty `/dev_host` as oci overlay dir and modified `install-docker.sh` to create the docker container in read-only mode (#2629).
- fix issue that the `S49chrony` init script could not startup if a NTP server was already added to the pool of NTP servers. Now a proper status code check is added and a warning will be output in case a certain NTP server could not be added to the server pool. (cf. https://homematic-forum.de/forum/viewtopic.php?f=65&t=80917).
- update buildroot to latest 2023.11.1
- add buildroot upstream patch to bump rpi wifi/bluetooth firmware to the latest version and also set the links correctly (#2566).
- fix ntpdate fails where we use echo with a newline to proceed correctly.
- integrate buildroot upstream patch to update linux-firmware to 20231030.
- modify docker healthcheck for oci platform so that rather than using monit for health status a plain startpFinished status check as well as a simple curl check is used (#2602).
- increase startup timeout limit of oci/docker platform including the HomeAssistant addon config to 240s (4 minutes) (#2604).
- rework `install-proxmox.sh` to allow virtual appliance installation on an ARM based Proxmox installation including selection of target disk size.
- update upstream kernel to 6.1.74.
- update java to 11.70.15-ca-jre11.0.22.
- update qemu-guest-agent to 8.2.0.
- update xe-guest-utilities to latest 8.4.0 version.
- update rpi4/rpi5 rpi-eeprom versions to latest versions.
- update rpi kernel and firmware to latest stable versions (stable_20240124).
- update tailscale to latest 1.58.2
- add linux kernel 6.4+ fix for `eq3_char_loop` kernel module compilation.
- add `/dev/null` redirection to omit any kill related error output when trying to kill psplash.
- foreground `irqbalance` to make sure to potentially not output debug info to syslog.
- fix ca-certificates local patch by removing the whole `/etc/ssl/certs` path.
- fix global lighttpd config to use 65535 for `server.max-request-field-size`.
- fix RaspberryPi Zero 2 W recognition.
- add `CONFIG_SERIAL_PRESENT` to `uboot.config` of ODROID boards to get uboot running correctly again (#2546).
- add patch for `wiringpi-rpi` package to skip the "Hardware" line check in `/proc/cpuinfo` which is not present anymore in the latest 6.1 kernel versions.
- fix `tailscaled` startup by fixing `PATH` to `/usr/sbin` only (#2541).

## 👪 Contributors (alphabetically):
- @Baxxy13 
- @jens-maus
- @jp112sdl 
- @MichaelN0815

## 📝 Support:
For support on installation and help please visit the following web pages:

[RaspberryMatic - Documentation :us:](https://github.com/jens-maus/RaspberryMatic/wiki/en.Home)
[RaspberryMatic - Discussions :us:](https://github.com/jens-maus/RaspberryMatic/discussions)
[RaspberryMatic - Dokumentation :de:](https://github.com/jens-maus/RaspberryMatic/wiki)
[RaspberryMatic - Forum :de:](https://homematic-forum.de/forum/viewforum.php?f=65)

## 📦 Download:
The following installation archives can be downloaded for different hardware platforms. To verify their integrity a `sha256` checksum is provided as well. You can either upload these files using the WebUI-based update mechanism or unarchive them to e.g. flash the included `*.img` files on a fresh installation media (e.g. microSD card):

- RaspberryPi5 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-rpi5.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-rpi5.zip)<br/>SHA256: 50458f1d26e6aba0d9f947be423057a93be5411fa4ca1c295769fec02b378261</sub>

- RaspberryPi4 Model B, RaspberryPi Compute Module 4, RaspberryPi 400 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-rpi4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-rpi4.zip)<br/>SHA256: 7c31a37a42928933d6ddd8f43e0de74a7d24407e5c8f77ae2b8865f53e5c91b6</sub>

- [CCU3](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-CCU3), [ELV-Charly](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-ELV-Charly), RaspberryPi3 Model B+, RaspberryPi3 Model B, RaspberryPi3 Model A+, RaspberryPi Compute Module 3, RaspberryPi Compute Module 3 lite, RaspberryPi Zero 2 W – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-rpi3.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-rpi3.zip)<br/>SHA256: a48b26410c731d15fc8f4499d9b40378ef01edaa35937dab89885056b29f252a</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130-ccu3.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-ccu3.tgz) (only for initial CCU3 Firmware -> RaspberryMatic Upgrade)<br/> SHA256: 6ca6037af024bf8914f8cf841170f9516bad5ccb4ce14b778228e905220fab6a</sub>

- RaspberryPi2 Model B – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-rpi2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-rpi2.zip)<br/>SHA256: 7dca4374149b0c4240aa2eeda8b5d64496bf160804ea5243fb54572aa20a2a04</sub>

- RaspberryPi Zero W, RaspberryPi Zero, RaspberryPi Compute Module 1, RaspberryPi1 (A+/B+) – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-RaspberryPi)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-rpi0.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-rpi0.zip)<br/>SHA256: 4d6f27492ae3ea5e0e2c5e8d71c319435bb3640f9a5931047fff0ea979d1ead6</sub>

- TinkerBoard S, TinkerBoard – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Tinkerboard)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-tinkerboard.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-tinkerboard.zip)<br/>SHA256: 12683defeb005c2769b6ad1fbb45311d18eb0ad32d73668125058bfcfe816a5a</sub>

- ODROID-N2/N2+/C4/C2 – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Odroid)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-odroid-n2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-odroid-n2.zip)<br/>SHA256: c74a1ac30bc2016b8d1b6bd7a79ff0ebf6a6026f5ebe26d72e0a0b53b9b0a57b</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130-odroid-c4.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-odroid-c4.zip)<br/>SHA256: d10de27c7da39c934760dc8813868417653b162350f7bb07d3f52f6694661a00</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130-odroid-c2.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-odroid-c2.zip)<br/>SHA256: 1fe59297f3afbe92d0f05437210a6fc6ca2a0ccbb0fe0f1ba6deaa2ff0db52a0</sub>

- Intel NUC – ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-IntelNUC)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-intelnuc.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-intelnuc.zip)<br/>SHA256: bb1b69c3d8dcc73a895590874f340f62d3c5fb37fe932fc19b8cc9d3dce4d1d1</sub>

- Open Virtual Appliance (OVA) – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE), [VirtualBox](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-VirtualBox), [ESXi](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-ESXi), [Synology](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Synology-VMM), [QNAP](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QNAP-VirtualizationStation), [Workstation Player](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-vmWare-Workstation-Player), [QEmu](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-QEmu), [UNRAID](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-UNRAID), [HyperV](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HyperV)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-ova.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-ova.zip)<br/>SHA256: 68cb3892b490b198fc47667dbe944f8bdaa53b30dd4451d38949b0671f70da0f</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130.ova](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130.ova) (only for initial OVA installation)<br/>SHA256: 9be673d90d1012e8233064bc6f7d9c482061da8281ed426eb2ac8eb4386359c4</sub>

- Docker / Open Container Initiative (OCI) – virtual appliance ([installation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-oci_amd64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-oci_amd64.tgz) (amd64/x86_64)<br/>SHA256: d62149cebafdc42e90d19e03480c45bd7f51121fdc8ff01f68214bc026b02f84</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130-oci_arm64.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-oci_arm64.tgz) (arm64/aarch64)<br/>SHA256: 49e053aed4ed39507649b0cbd42f6e32a66ed7acbe197d889bd6b26da17f286e</sub>
<sub>📦 [RaspberryMatic-3.73.9.20240130-oci_arm.tgz](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-oci_arm.tgz) (arm/armhf)<br/>SHA256: a774fd0c97ea4cc373cd703e0ee6421a39a40516c8fe8e4f7bcc25961b83ce63</sub>

- Generic-aarch64 Appliance – ([ProxmoxVE](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Proxmox-VE)):
<sub>📦 [RaspberryMatic-3.73.9.20240130-generic-aarch64.zip](https://github.com/jens-maus/RaspberryMatic/releases/download/3.73.9.20240130/RaspberryMatic-3.73.9.20240130-generic-aarch64.zip)<br/>SHA256: 3a2958783dbf2e0a7763319bf2f61db55238330216fc39d5e432ff11fbe3584d</sub>

- Kubernetes / K8s – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Kubernetes)</sub>

- Home Assistant Add-on – virtual appliance:
<sub>see [install documentation](https://github.com/jens-maus/RaspberryMatic/wiki/Installation-HomeAssistant)</sub>

<hr /><em>This discussion was created from the release <a href='https://github.com/jens-maus/RaspberryMatic/releases/tag/3.73.9.20240130'>RaspberryMatic 3.73.9.20240130</a>.</em>

**Comment by Tobix99** [comment]
Hi, thanks for the release! Short question: does the RPi5 support also cover RPi5 with Docker? As if I start the docker on my Pi5, I get a 139 Exit Code: `Exited (139)`. Is this a Bug or is this still expected?

Example command: `sudo docker run ghcr.io/jens-maus/raspberrymatic:3.73.9.20240130`

**Reply by jens-maus** [reply]
This *should* work, thought haven't tested it myself yet ;)

**Reply by Tobix99** [reply]
Mhhh ok, I've tried the same exact command on my RPi4 and there I get at least some output ... I don't expect it to boot up correctly, but I think it should at least output some logs ... but maybe it's a problem with my setup, I'll check. Thanks for the quick reply :)

RPi4
```
sudo docker run ghcr.io/jens-maus/raspberrymatic:3.73.9.20240130
Unable to find image 'ghcr.io/jens-maus/raspberrymatic:3.73.9.20240130' locally
3.73.9.20240130: Pulling from jens-maus/raspberrymatic
fca4fef8d854: Pull complete 
Digest: sha256:ffe33510096319528e7d97a0cefe48bf1a80688015c99b558eff1a5abd94a888
Status: Downloaded newer image for ghcr.io/jens-maus/raspberrymatic:3.73.9.20240130
mount: permission denied (are you root?)
Identifying host system: Raspberry Pi 4 Model B Rev 1.1 (oci), OK
....
```

**Reply by Tobix99** [reply]
Hi, I finally found the problem. It was related to the 16k vs 4k page sizes. I hadn't made any changes regarding the page sizes on the RPi5, and that's why the container was always exiting with code 139 (memory violation issue). Perhaps this should be mentioned on the Docker installation page? To check what current page size you are using you can use `getconf PAGESIZE`.

Here's the link that gave me the idea to change the page size:
https://github.com/home-assistant/operating-system/discussions/2844#discussioncomment-7738971

**Reply by jens-maus** [reply]
Yes, unfortunately the RaspberryPi / RaspberryPiOS makers decided to switch from 4k memory pages to 16k memory pages due to minor performance improvements. This however overcomplicates a lot of things as you now noticed in your case. While with RaspberryPiOS you can switch to a 4k kernel this is currently for example not possible with Armbian or other distributions. Thus, more of these 4k vs. 16k memory pages issues will come up once the RaspberryPi5 will be more widely used....

However, in RaspberryMatic on RapsberryPi5 I decided to stay with 4k memory pages to stay compatible as much as possible.

**Reply by Tobix99** [reply]
I couldn't locate information about the 16k problem in the wiki. It might be a valuable addition to mention it there, perhaps within the docker section at the top where the highlighted "Important" text box is?

https://github.com/jens-maus/RaspberryMatic/wiki/Installation-Docker-OCI#known-limitations

**Reply by jens-maus** [reply]
Feel free to add it yourself as this a freely editable wiki for everyone to contribute.

**Reply by jens-maus** [reply]
@Tobix99 Please try again with a 16k enabled kernel and the latest nightly snapshot of the RaspberryMatic docker container. I think I found a solution for the issues with 16k kernels and RaspberryPi5. Thus, the latest nightly snapshot docker container should work with a standard 16k kernel again on the RaspberryPi5.

**Reply by Tobix99** [reply]
Hi, yes I think it worked. I just started the container without any mounts, just the `docker run` on the 16k Pagesize an it booted up just fine 👍 . But I will revert to 4k as the "official" release it not 16k compatible to I can't test everything sadly.

Log:
```
docker run ghcr.io/jens-maus/raspberrymatic:3.73.9.20240204-e76ff9d
Unable to find image 'ghcr.io/jens-maus/raspberrymatic:3.73.9.20240204-e76ff9d' locally
3.73.9.20240204-e76ff9d: Pulling from jens-maus/raspberrymatic
0d84282aa5a5: Pull complete 
Digest: sha256:2efef65b826f4c792eb5677ed573ba2b51ff973e1d945b77f47b41511b6a8ad7
Status: Downloaded newer image for ghcr.io/jens-maus/raspberrymatic:3.73.9.20240204-e76ff9d
mount: permission denied (are you root?)
Identifying host system: Raspberry Pi 5 Model B Rev 1.0 (oci), OK
Initializing RTC Clock: no hardware found
Running sysctl: OK
Checking for Factory Reset: not required
Checking for Backup Restore: not required
Running seedrng: OK
Initializing System: mkdir: can't create directory '/var/lock': No such file or directory
mount: permission denied (are you root?)
OK
Setup ca-certificates: OK
Starting logging: OK
Populating /dev using udev: starting version 3.2.14
done
Init onboard LEDs: init, OK
Starting iptables: OK
Starting network: eth0: link up, fixed, firewall, inet up, 172.17.0.2, OK
Identifying Homematic RF-Hardware: HmRF: n/a, HmIP: n/a, OK
Updating Homematic RF-Hardware: no GPIO/USB connected RF-hardware found
Starting hs485dLoader: disabled
Starting xinetd: OK
Starting eq3configd: OK
Starting lighttpd: creating new SSL cert... OK
Starting ser2net: disabled
Starting ssdpd: OK
Starting NUT services: disabled
Initializing Third-Party Addons: OK
Starting LGWFirmwareUpdate: ...OK
Setting LAN Gateway keys: OK
Starting hs485d: disabled
Starting multimacd: not required
Starting rfd: no BidCos-RF hardware found
Starting HMIPServer: ...OK
Starting ReGaHss: .OK
Starting CloudMatic: OK
Starting NeoServer: OK
Starting Third-Party Addons: OK
Starting crond: OK
Setup onboard LEDs: booted, OK
Finished Boot: 3.73.9.20240204 (raspmatic_oci_arm64)
^CRebooting container
Setup onboard LEDs: shutdown, OK
Stopping crond: OK
Stopping Third-Party Addons: OK
Stopping NeoServer: OK
Stopping ReGaHss: ....OK
Stopping HMIPServer: OK
Stopping rfd: OK
Stopping multimacd: not started
Stopping hs485d: OK
Stopping NUT services: OK
Stopping sshd: OK
Stopping ssdpd: OK
Shutting down ser2net: OK
Stopping lighttpd: OK
Stopping eq3configd: OK
Stopping xinetd: OK
Stopping network: eth0: OK
Stopping iptables: iptables v1.8.9 (legacy): can't initialize iptables table `filter': Permission denied (you must be root)
Perhaps iptables or your kernel needs to be upgraded.
FAIL
Stopping logging: OK
Cleaning up System: OK
Running seedrng: seedrng: RNDADDENTROPY: Operation not permitted
OK
Cleaning up RTC Clock: no hardware found
The system is going down NOW!
Sent SIGTERM to all processes
Sent SIGKILL to all processes
Requesting system reboot
```

**Reply by Tobix99** [reply]
Hi @jens-maus, I wanted to report back, that I'm using the Snapshot Docker-Image for 3 Days on a 16K Pagesize Raspberry Pi 5 🎉. I've have no problems so far, so the fix is working.

`ghcr.io/jens-maus/raspberrymatic:3.75.6.20240307-2d3b4b6`
